crypto/evp/p_rsa.c

```diff
@@ -96,6 +99,78 @@ typedef struct {
   size_t len;
 } RSA_OAEP_LABEL_PARAMS;
 
+static int pkey_ctx_is_pss(EVP_PKEY_CTX *ctx) {
+  return ctx && ctx->pmeth && (ctx->pmeth->pkey_id == EVP_PKEY_RSA_PSS);
```

Comment:
In a very unlikely case that the pointers are such that ctx = ~ctx->pmeth, the function will incorrectly return 0. I think it's better to check explicitly:
`(ctx != NULL) && (ctx->pmeth != NULL) && ...`